src/machine/machine_rp2_2040.go

```diff
@@ -113,6 +113,19 @@ const (
 	fnXIP pinFunc = 0
 )
 
+// System clock configuration
+const (
+	pllSysFreq     uint32 = 125 * MHz
```

Comment:
The rp2040 datasheet says that its cores are capable of 133 MHz. A cursory glance at the datasheet didn't mention any downside to that speed, so why is TinyGo running at 125 MHz? Perhaps it's because of non-integer periods that you mention in the PR description for rp2350? If so, this PR might consider rp2040 at 133 MHz for the same reasons you chose 150 MHz for rp2350.

Failing all that, a comment would be nice.

Reply:
That's a good point, I don't know why it was running at 125 either, but this seems plausible.